code/game/objects/items/devices/flashlight.dm

```diff
@@ -147,14 +150,76 @@
 
 /obj/item/device/flashlight/pen
 	name = "penlight"
-	desc = "A pen-sized light, used by medical staff."
+	desc = "A pen-sized light, used by medical staff to check the condition of eyes, brain and cosncience of patients."
 	icon_state = "penlight"
 	item_state = ""
+	flags_equip_slot = SLOT_WAIST|SLOT_EAR|SLOT_SUIT_STORE
 	flags_atom = FPRINT|CONDUCT
 	light_range = 2
 	w_class = SIZE_TINY
+	throw_speed = SPEED_VERY_FAST
+	throw_range = 15
+	matter = list("metal" = 10,"glass" = 5)
 	raillight_compatible = 0
 
+/obj/item/device/flashlight/pen/attack(mob/living/carbon/human/M as mob, mob/living/user as mob)
+	add_fingerprint(user)
+	if(on && user.zone_selected == "eyes")
+		if(istype(M, /mob/living/carbon/human)) //robots and aliens are unaffected
+			var/datum/internal_organ/eyes/eyes = M.internal_organs_by_name["eyes"]
+			var/datum/internal_organ/brain/brain = M.internal_organs_by_name["brain"]
+			var/reaction = "try to watch closely, but you see no difference in their eyes' reaction"
+			if(isnull(M.internal_organs_by_name))
+				reaction = "discover that indeed he has nothing to check"
+				return // they have no organs somehow
+			if(M == user) //they're using it on themselves
+				M.flash_eyes()
+				M.visible_message(SPAN_NOTICE("[M] directs [src] to \his eyes."), \
```

Comment:
we have a gender macro that works better than \his, see line 188

Reply:
I know, that is still part of the older flashlight flashing code, I didn't replace it on the philosophy that you shouldn't be replacing stuff that is working, but I can easily replace it with a "[M.p_their()]", no problems at all in that

Reply:
Yes, do replace it

Reply:
resolved